### ISSUE TITLE
UseDataClass false positive

### ISSUE THREAD

**Issue by igorwojda** [arturbosch/detekt#1585]
A delegate is used to save data in Android shared preferences, so this class should not be a data class.

## Expected Behavior
Do not report this as a rule violation

## Observed Behavior
Reports as a rule violation

## Steps to Reproduce
```
class SharedPreferenceRepository(
    context: Context
) {
    companion object {
        private const val PREFIX = "APP"
        private const val INSTALL_UUID: String = "${PREFIX}_INSTALL_UUID"
    }

    private var _installUUID: String by SharedPreferencesExtension(context, INSTALL_UUID, "")
}
```

**Comment by schalkms** [comment]
@igorwojda How should the rule detect that the mentioned class is no data class candidate?
Are there any indicators?
